models/articles_models.js

```diff
@@ -46,20 +46,64 @@ exports.patchArticleById = (inc_votes, article_id) => {
     });
 };
 
-exports.fetchArticles = () => {
-  return connection
-    .query(
-      `SELECT articles.*, COUNT(comments.article_id)::INT AS comment_count 
-    FROM articles 
-    LEFT JOIN comments 
-    ON comments.article_id = articles.article_id
-    GROUP BY articles.article_id
-    ORDER BY created_at DESC`
-    )
-    .then((articles) => {
-      return articles.rows;
-    })
-    .catch((err) => {
-      return err;
+exports.fetchArticles = async (
+  sort_by = "created_at",
+  order = "desc",
+  topic
+) => {
+  const validSort_by = [
+    "topic",
+    "title",
+    "author",
+    "body",
+    "created_at",
+    "votes",
+  ];
+
+  const validOrder = ["ASC", "DESC", "asc", "desc"];
+
+  if (!validSort_by.includes(sort_by)) {
+    return Promise.reject({
+      status: 400,
+      msg: `cannot sort by ${sort_by}`,
+    });
+  }
+
+  if (!validOrder.includes(order)) {
+    return Promise.reject({
+      status: 400,
+      msg: `cannot sort by ${order}`,
     });
+  }
+
+  const topicValidate = await connection.query(
+    `SELECT * FROM topics WHERE slug = $1`,
+    [topic]
+  );
+
+  if (topic) {
+    if (topicValidate.rowCount === 0) {
+      return Promise.reject({
+        status: 400,
+        msg: `cannot sort by ${topic}`,
+      });
+    } else {
+      return connection
+        .query(
+          `SELECT articles.*, COUNT(comments.article_id)::INT AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id WHERE topic = $1 GROUP BY articles.article_id ORDER BY ${sort_by} ${order}`,
+          [topic]
+        )
```

Comment:
would be nice to format this on multiple lines like you have your previous PSQL queries